### PR TITLE
Modify CVE-2023-1177 to cover CVE-2023-6977

### LINKED ISSUE #450: AI PRP: Arbitrary File Read in mlflow CVE-2023-6977
Hi,

I want to develop a plugin for mlflow LFI - CVE-2023-6977.

Vulnerability Information: This vulnerability enables malicious users to read sensitive files on the server.

Vulnerable Versions are below the 2.9.2

References:
- https://nvd.nist.gov/vuln/detail/CVE-2023-6977
- https://github.com/mlflow/mlflow/commit/4bd7f27c810ba7487d53ed5ef1038fca0f8dc28c
- https://huntr.com/bounties/fe53bf71-3687-4711-90df-c26172880aaf

The vulnerability requires three HTTP requests one is GET and the other two are POST. After creating a model, by using a symlink, one can read files on the filesystem.

Actually, after sending an AI PRP for a CVE that already exists on the Tsunami, I searched this CVE code on the repository checked the issues, and found nothing. I hope it is not found in the Tsunami and this is not a duplicated issue.

### PR DESCRIPTION
**Pull request by frkngksl**
Hi @tooryx ,

This is the modification that resolves #450 

Vulnerable and Fixed Environments are here: https://github.com/google/security-testbeds/pull/61

### CODE REVIEW

**Comment by tooryx** [comment]
Thank you @frkngksl, this looks good overall. I will take some time to test it soon.

_~tooryx_

**Comment by frkngksl** [comment]
Thanks a lot @tooryx !

**Comment by tooryx** [comment]
Could you please fill the related field with all three CVEs? Here is [an example](https://github.com/google/tsunami-security-scanner-plugins/blob/e7cbb377445c80ed12e33077e3c4aacf4abcbe26/community/detectors/jenkins_arbitrary_file_read_cve_2024_23897/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202423897/Cve202423897Detector.java#L243)

File: community/detectors/mlflow_cve_2023_6977/src/test/java/com/google/tsunami/plugins/detectors/cves/cve20236977/Cve20236977DetectorTest.java [hunk 1]
```diff
@@ -106,12 +106,12 @@ public void detect_whenVulnerable_returnsVulnerability() throws IOException {
...
                                 .setPublisher("TSUNAMI_COMMUNITY")
-                                .setValue("CVE_2023_1177"))
+                                .setValue("CVE_2023_6977"))
                         .setSeverity(Severity.CRITICAL)
-                        .setTitle("CVE-2023-1177 MLflow LFI/RFI")
+                        .setTitle("CVE-2023-6977 MLflow LFI/RFI")
                         .setRecommendation(
-                            "1.Updated to version 2.2.1 or later\n2.Add authentication to MLflow "
-                                + "server\n")
+                            "1.Updated to any version above the version 2.9.2\n"
```
Comment (tooryx) [inline community/detectors/mlflow_cve_2023_6977/src/test/java/com/google/tsunami/plugins/detectors/cves/cve20236977/Cve20236977DetectorTest.java:113]: Please make the fixed version 2.10.0. When testing locally, it is obvious that the vulnerability has not been patched, even if marked as such by both the advisory and bug tracker.

**Comment by tooryx** [comment]
> > Could you please fill the relatedId field with all three CVEs? Here is [an example](https://github.com/google/tsunami-security-scanner-plugins/blob/e7cbb377445c80ed12e33077e3c4aacf4abcbe26/community/detectors/jenkins_arbitrary_file_read_cve_2024_23897/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202423897/Cve202423897Detector.java#L243)
> 
> Done @tooryx !

Thank you! Although, you need to add the 3 of them. This field should contain ALL related CVEs

**Comment by tooryx** [comment]
Looks good, thanks. I expect it should be merged next week.

**Comment by tooryx** [comment]
Hi @frkngksl,

Your PR has been merged. This usually means a reward will be granted. Google will start the internal QC process and the reward amount will be determined based on the quality of the detector report. Please be patient and allow up to a week for the QC process to finish. You'll be notified once the decision is made.

Thanks!

**Comment by tooryx** [comment]
We are most likely going to review it this week.

_~tooryx_

**Pull request merged**